hibernate-core/src/main/java/org/hibernate/sql/ast/spi/SimpleFromClauseAccessImpl.java

```diff
@@ -75,6 +75,18 @@ public TableGroup findTableGroupForGetOrCreate(NavigablePath navigablePath) {
 		return tableGroup;
 	}
 
+	private TableGroup locateTableGroup(NavigablePath navigablePath) {
+		final TableGroup tableGroup = tableGroupMap.get( navigablePath );
+		if ( tableGroup != null || parent == null ) {
+			return tableGroup;
+		}
+		final NavigablePath navigablePathParent = navigablePath.getParent();
+		if ( navigablePathParent == null || tableGroupMap.get( navigablePathParent ) == null ) {
```

Comment:
I think this is actually not so easy. We also have to consider grandparents etc.

Reply:
Consider a slightly more complicated query like this:

```
select
 e.embeddable.otherEntity.id," +
 (select e.aString from MyEntity e where e.embeddable.otherEntity.id = :anotherEntityId)
from MyEntity e where e.id = :id
```

Reply:
I was thinking we could generalize the check in `findTableGroupForGetOrCreate` which cycles to find the very first parent to verify all table groups and not just embeddables.